### ISSUE TITLE
[A11yAudit] Stepper carousel is not accessible - (Stepper API work)

### ISSUE THREAD

**Issue by ssabo1** [sparkdesignsystem/Backlog#1320]
**User Story:**
As Spark we want to ...

**Notes:**
Paciello Audit on Spark Design System
QLC-21 Tab Panel is what Spark calls Stepper

Pacello:
In general, we would discourage the use of carousel widgets. The primary reasons are outlined in Should I use a carousel?. However, carousels can be made reasonably accessible provided that general accessibility concepts are correctly implemented. Regarding the image carousel that is revealed at the mobile breakpoint, this appears to be structured appropriately, is correctly exposed to assistive technologies, and can be focused and operated using the keyboard. The extraneous tab panel markup, however, should be removed when not in use.

For an example implementation of a carousel, we recommend following the guidelines on Carousel Concepts - WAI Web Accessibility Tutorials.

**RECOMMENDATIONS**
The hybrid of a tab panel and image carousel, with different components exposed at different breakpoints, results in a confusing and inaccessible experience for assistive technology users. Furthermore, although the appropriate tab panel markup has been used, the widget does not conform to a standard tab panel pattern, which may explain the inconsistent behavior exhibited by screen readers. As a result, we recommend either separating the widgets completely and displaying one or the other depending on the breakpoint, or marking the image carousel up as a tab panel to main consistency between the two breakpoints.

Regarding the tab panel, the container for the tabs correctly has a role of tablist (although it lacks an accessible name describing its purpose), and the tabs in the list have a role of tab. The tabindex attribute has also been used to ensure the tabs can receive keyboard focus. Although the aria-selected attribute is applied correctly, the aria-controls attribute is wrongly applied to a <span> element within the tab, instead of to the tab itself.

The panels that are controlled by the tabs correctly have a role of tabpanel. However, the aria-labelledby attribute on each tab panel points not at the tab but at the <span> element within the tab. Visibility of the tab panels appears to be controlled correctly with the display:none; and visibility:hidden; CSS properties. The tab panels, however, are nested within the tabs. Also, the tab panels lack a tabindex attribute, which would facilitate movement to the panel content for assistive technology users. We recommend restructuring the tab panel widget in accordance with an established tab panel pattern. This could also be adapted to be displayed at the mobile breakpoint (albeit without the carousel controls).

**Comment by ssabo1** [comment]
Confirmed: This has been added to Spark Design System Team Backlog.
